### PR TITLE
Disable parallel-compiler in CI

### PR DESCRIPTION
**Pull request by Veykril**

### CODE REVIEW

**Review by japaric — APPROVED** [review]

**Comment by Veykril** [comment]
bors try

**Comment by Veykril** [comment]
bors cancel

**Comment by Veykril** [comment]
bors try

File: src/bootstrap/src/lib.rs [hunk 1]
```diff
@@ -721,7 +719,7 @@ impl Build {
         if self.config.rustc_parallel && check("rustc_use_parallel_compiler") {
             features.push("rustc_use_parallel_compiler");
         }
-        if self.config.rust_randomize_layout {
+        if self.config.rust_randomize_layout && check("rustc_randomized_layouts") {
```
Comment (tshepang) [inline src/bootstrap/src/lib.rs:722]: what is `check` for here... it does not exist upstream

Comment (Veykril) [reply src/bootstrap/src/lib.rs:722]: it does, its a closure a couple lines up. It landed in the same rollup as the `rustc_randomize_layout` config being added hence the discrepancy.

**Comment by Veykril** [comment]
Good failure, this is progress!

**Comment by Veykril** [comment]
bors try

File: src/bootstrap/src/lib.rs [hunk 2]
```diff
@@ -700,15 +700,13 @@ impl Build {
...
             .iter()
             .flat_map(|krate| &self.crates[krate].features)
             .map(std::ops::Deref::deref)
             .collect();
-        let check = |_feature: &str| -> bool {
-            //crates.is_empty() || possible_features_by_crates.contains(feature)
-            // change necessary while resolver is broken
-            true
+        let check = |feature: &str| -> bool {
+            crates.is_empty() || possible_features_by_crates.contains(feature)
```
Comment (Urhengulas) [inline src/bootstrap/src/lib.rs:709]: `possible_features_by_crates` is constructed from `crates`, so iiuc `possible_features_by_crates` will be empty if `crates` is empty, correct?

Also `crates` is immutable.

If above is true, then it would be enough to only do `possible_features_by_crates.contains(feature)`, because if `possible_features_by_crates` is empty it will not contain `feature`.

File: compiler/rustc_index_macros/Cargo.toml [hunk 3]
```diff
@@ -7,7 +7,7 @@ edition = "2021"
 proc-macro = true
 
 [dependencies]
-syn = { version = "2.0.9", features = ["full"] }
+syn = { version = "2.0.9", features = ["full", "extra-traits"] }
```
Comment (Urhengulas) [inline compiler/rustc_index_macros/Cargo.toml:10]: Why do you need to enable the feature here and in `compiler/rustc_type_ir_macros/Cargo.toml`? Did it work with resolver 1 but not resolver 2?

File: compiler/rustc_transmute/src/maybe_transmutable/tests.rs [hunk 4]
```diff
@@ -107,7 +107,7 @@ mod bool {
             #[cfg(feature = "rustc")]
             let mut set = crate::Set::default();
             #[cfg(not(feature = "rustc"))]
-            let mut set = std::collections::HashSet::new();
+            let mut set = rustc_data_structures::fx::FxHashSet::default();
```
Comment (Urhengulas) [inline compiler/rustc_transmute/src/maybe_transmutable/tests.rs:110]: Also how is this change related?

**Review by Urhengulas — CHANGES_REQUESTED** [review]

File: compiler/rustc_index_macros/Cargo.toml (hunk 3 above)
Comment (Veykril) [reply compiler/rustc_index_macros/Cargo.toml:10]: Due to how resolver 2's feature unification works, resolver 1 always unifies across the workspace, resolver 2 only across the packages being built (which is the entire problem we were having as upstream has set up the features incorrectly).

This change will be upstreamed afterwards.

File: compiler/rustc_transmute/src/maybe_transmutable/tests.rs (hunk 4 above)
Comment (Veykril) [reply compiler/rustc_transmute/src/maybe_transmutable/tests.rs:110]: rustc runs clippy that lints against using bare `HashSet`, upstream doesnt catch this because they dont build without the `rustc` feature.

This change will be upstreamed afterwards.

File: src/bootstrap/src/lib.rs (hunk 2 above)
Comment (Veykril) [reply src/bootstrap/src/lib.rs:709]: This change brings this code back in sync with upstream.

Also those do not mean the same thing, empty `possible_features_by_crates` means no crates have any features that are allowed, so the closure should always return false in that case, where as an empty `crates` means we will build all crates so any feature is fine to be passed.

> Also crates is immutable.

?

**Review by Urhengulas — APPROVED** [review]
bors merge

**Comment by Veykril** [comment]
bors cancel
I'll investigate the discussed alternative first

**Comment by Veykril** [comment]
Upstream removed the cfg entirely
